app/controllers/products_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class ProductsController < ApplicationController
+  before_action :load_product, only: %i[show]
+  before_action :show_slider, only: %i[index show]
+  before_action :load_categories, only: %i[index]
+
+  def index
+    @pagy, @products = pagy(filter_products, items: Settings.PAGE_9)
+  end
+
+  def new
+    @product = Product.new
+  end
+
+  def create
+    product = Product.new(product_params)
+    if product.save
+      flash[:success] = I18n.t("products.create.success")
+      redirect_to(products_path)
+    else
+      flash.now[:error] = I18n.t("products.create.error")
+      render(:new)
+    end
+  end
+
+  def show; end
+
+  private
+
+  def load_categories
+    @categories = Category.all.map { |category| [category.name, category.name] }
```

Comment:
```suggestion
    @categories = Category.pluck(:name, id).map { |category| [category.name, category.name] }
```

ủa sao show name và name thế, ko phải name và id à

Reply:
Dạ em đã sửa lại truyền id cho params rồi ạ